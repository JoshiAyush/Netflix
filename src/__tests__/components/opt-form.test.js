import React from "react";
import { render } from "@testing-library/react";

import { OptFormContainer } from "../../components/index.js";

describe("<OptForm />", () => {
    it("renders the <OptForm /> with populated data", () => {
        const { container, getByText, getByPlaceholderText } = render(
            <OptFormContainer>

                <OptFormContainer.Input placeholder="Email address" />

                <OptFormContainer.Button width="32.8%">Try it now</OptFormContainer.Button>

                <OptFormContainer.Text>

                    Ready to watch? Enter your email to create or restart your membership

                </OptFormContainer.Text>

            </OptFormContainer>
        );

        expect(getByText("Try it now")).toBeTruthy();
        expect(getByText("Ready to watch? Enter your email to create or restart your membership")).toBeTruthy();
        expect(getByPlaceholderText("Email address")).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });
});