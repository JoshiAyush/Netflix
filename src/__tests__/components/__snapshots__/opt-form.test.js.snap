// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders the <OptForm /> with populated data 1`] = `
<div
  class="style__Container-aw8769-0 emHGMr"
>
  <input
    class="style__Input-aw8769-1 jcPedt"
    placeholder="Email address"
  />
  <button
    class="style__Button-aw8769-2 jJrQMn"
    width="32.8%"
  >
    Try it now
     
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <p
    class="style__Text-aw8769-3 dicfXI"
  >
    Ready to watch? Enter your email to create or restart your membership
  </p>
</div>
`;
